drivers/sensor/renesas/rrh46410/rrh46410.c

```diff
@@ -0,0 +1,301 @@
+	int rc;
+	uint8_t checksum = ~RRH46410_GET_OPERATION_MODE;
+	uint8_t operation_mode;
+	uint8_t get_operation[2];
+
+	get_operation[0] = RRH46410_GET_OPERATION_MODE;
+	get_operation[1] = checksum;
+	rc = data->hw_tf->write_data(dev, get_operation, sizeof(get_operation));
+
+	if (rc < 0) {
+		LOG_ERR("Failed to send get.");
+		return rc;
+	}
+
+	rc = data->hw_tf->read_data(dev, data->read_buffer, 2);
+	if (rc < 0) {
+		LOG_ERR("Failed to read data from device.");
+		return rc;
+	}
+
+	operation_mode = data->read_buffer[1];
+
+	return operation_mode;
+}
+
+static int rrh46410_set_operation_mode(const struct device *dev)
+{
+	struct rrh46410_data *data = dev->data;
+	int rc;
+	uint8_t checksum = ~(RRH46410_SET_OPERATION_MODE + RRH46410_OPERATION_MODE_IAQ_2ND_GEN);
+	uint8_t set_operation[3];
+
+	set_operation[0] = RRH46410_SET_OPERATION_MODE;
+	set_operation[1] = RRH46410_OPERATION_MODE_IAQ_2ND_GEN;
+	set_operation[2] = checksum;
+
+	rc = data->hw_tf->write_data(dev, set_operation, sizeof(set_operation));
+
+	if (rc < 0) {
+		LOG_ERR("Failed to send set.");
+		return rc;
+	}
+
+	return 0;
+}
+
+static int rrh46410_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	struct rrh46410_data *data = dev->data;
+	double val_rrh46410 = sensor_value_to_double(val);
+	uint8_t encoded_humidity = (uint8_t)((val_rrh46410 / 100) * 255);
+	uint8_t command_data = encoded_humidity;
+	uint8_t checksum = ~(RRH46410_SET_HUMIDITY + command_data);
+	uint8_t set_humidity[3];
+	int rc;
+
+	set_humidity[0] = RRH46410_SET_HUMIDITY;
+	set_humidity[1] = command_data;
+	set_humidity[2] = checksum;
+
+	if (chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_RRH46410_HUMIDITY:
+		rc = data->hw_tf->write_data(dev, set_humidity, sizeof(set_humidity));
+		if (rc < 0) {
+			LOG_ERR("Failed to send humidity.");
+			return rc;
+		}
+
+		k_msleep(10);
+
+		rc = data->hw_tf->read_data(dev, data->read_buffer, 1);
+		if (rc < 0) {
+			LOG_ERR("Failed to read data from device.");
+			return rc;
+		}
+
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int rrh46410_read_sample(const struct device *dev, uint8_t *sample_counter,
+				uint8_t *iaq_sample, uint16_t *tvoc_sample, uint16_t *etoh_sample,
+				uint16_t *eco2_sample, uint8_t *reliaq_sample)
+{
+	struct rrh46410_data *data = dev->data;
+	uint8_t status;
+	int rc;
+
+	rc = data->hw_tf->read_data(dev, data->read_buffer, sizeof(data->read_buffer));
+	if (rc < 0) {
+		LOG_ERR("Failed to read data from device.");
+		return rc;
+	}
+
+	status = data->read_buffer[0];
+	*sample_counter = data->read_buffer[1];
+	*iaq_sample = data->read_buffer[2];
+	*tvoc_sample = sys_get_be16(&data->read_buffer[3]);
+	*etoh_sample = sys_get_be16(&data->read_buffer[5]);
+	*eco2_sample = sys_get_be16(&data->read_buffer[7]);
+	*reliaq_sample = sys_get_be16(&data->read_buffer[9]);
```

Comment:
reliaq_sample is a uint8_t?

```suggestion
	*reliaq_sample = data->read_buffer[9];
```